functions/sync_images/main.py

```diff
@@ -0,0 +1,148 @@
+
+    attachment_service = AttachmentService(storage_client, get_requests_session(**request_retry_options))
+    publish_service = PublishService(TOPIC_NAME, get_requests_session(**request_retry_options))
+
+    # Getting all form blobs
+    form_blobs = bucket.list_blobs(prefix=ENTRY_FILEPATH_PREFIX + form_storage_suffix)
+
+    result = {
+        "total_form_count": 0,
+        "form_with_missing_attachment_count": 0,
+        "missing_attachment_count": 0,
+        "downloaded_attachment_count": 0
+    }
+
+    # Looping through all forms to check them.
+    for form_blob in form_blobs:
+        result["total_form_count"] += 1
+        form_data = json.loads(form_blob.download_as_string())
+
+        try:
+            form = Form(form_data)
+        except KeyError as exception:
+            logging.error(
+                f"Invalid form: {form_blob.name}\n"
+                f"Exception: {str(exception)}"
+            )
+            continue
+
+        # Find all a form's attachments that are not available in storage.
+        missing_attachments = attachment_service.find_missing_attachments(form)
+
+        if missing_attachments:
+            missing_attachment_count = len(missing_attachments)
+            result["form_with_missing_attachment_count"] += 1
+            result["missing_attachment_count"] += missing_attachment_count
+
+            if not skip_download:
+                logging.info(
+                    f"Found {missing_attachment_count} missing attachments for {form_blob.name}, "
+                    "attempting to download..."
+                )
+
+                for attachment in missing_attachments:
+                    # Downloading attachment to storage.
+                    success, response = attachment_service.download(attachment)
+                    if success:
+                        result["downloaded_attachment_count"] += 1
+                    else:
+                        logging.error(
+                            "Error downloading image.\n"
+                            f"Form: {form_blob.name}\n"
+                            f"URL: {attachment.download_url}\n"
+                            f"Bucket path: {attachment.bucket_path}\n"
+                            f"Response: {response}"
+                        )
+
+                logging.info("Download(s) complete.")
+
+        if (missing_attachments and not skip_download) or force_arcgis_update:
+            logging.info("Sending form to ArcGIS...")
+
+            # Sending the form to ArcGIS
+            publish_service.publish_form(form)
+
+    return json.dumps(result), 200
+
+
+def get_request_arguments(request):
+    """
+    Extracts all arguments from HTTP arguments and JSON body.
+
+    :param request: The request object to extract from.
+    :type request: flask.Request
+
+    :return: A dictionary of all arguments.
+    :rtype: dict
+    """
+
+    arguments = dict()
+    if not request:
+        return arguments
```

Comment:
one place to return the expected value of a function is recommended. why not use:

if request: and remove the return